model/state_model_nlparam.jl

```diff
@@ -125,43 +126,43 @@ cge = MCPModel();
 @NLparameter(cge, theta_n[r in regions, g in goods] == value(nd0[r, g]) / (value(nd0[r, g]) - value(dd0[r, g])));
 @NLparameter(cge, theta_m[r in regions, g in goods] == (1+value(tm0[r, g])) * value(m0[r, g]) / (value(nd0[r, g]) + value(dd0[r, g]) + (1 + value(tm0[r, g])) * value(m0[r, g])));
 
-replace_nan_inf(alpha_kl)
+replace_nan_inf(alpha_kl) 
 replace_nan_inf(alpha_x)
 replace_nan_inf(alpha_d)
 replace_nan_inf(alpha_n)
 replace_nan_inf(theta_n)
 replace_nan_inf(theta_m)
 
 #Substitution and transformation elasticities
-@NLparameter(cge, es_va[r in regions, s in sectors] == 1); # value-added nest - substitution elasticity
-@NLparameter(cge, es_y[r in regions, s in sectors] == 0); # Top-level Y nest (VA,M) - substitution elasticity
-@NLparameter(cge, es_m[r in regions, s in sectors] == 0); # Materials nest - substitution elasticity
-@NLparameter(cge, et_x[r in regions, g in goods] == 4); # Disposition, distribute regional supply to local, national, export - transformation elasticity
-@NLparameter(cge, es_a[r in regions, g in goods] == 0); # Top-level A nest for aggregate demand (Margins, goods) - substitution elasticity
-@NLparameter(cge, es_mar[r in regions, g in goods] == 0); # Margin supply - substitution elasticity
-@NLparameter(cge, es_d[r in regions, g in goods] == 2); # Domestic demand aggregation nest (intranational) - substitution elasticity
-@NLparameter(cge, es_f[r in regions, g in goods] == 4); # Domestic and foreign demand aggregation nest (international) - substitution elasticity
+@NLparameter(cge, es_va[r in regions, s in sectors] == SUB_ELAST[:va]); # value-added nest - substitution elasticity
+@NLparameter(cge, es_y[r in regions, s in sectors]  == SUB_ELAST[:y]); # Top-level Y nest (VA,M) - substitution elasticity
+@NLparameter(cge, es_m[r in regions, s in sectors]  == SUB_ELAST[:m]); # Materials nest - substitution elasticity
+@NLparameter(cge, et_x[r in regions, g in goods]    == TRANS_ELAST[:x]); # Disposition, distribute regional supply to local, national, export - transformation elasticity
+@NLparameter(cge, es_a[r in regions, g in goods]    == SUB_ELAST[:a]); # Top-level A nest for aggregate demand (Margins, goods) - substitution elasticity
+@NLparameter(cge, es_mar[r in regions, g in goods]  == SUB_ELAST[:mar]); # Margin supply - substitution elasticity
+@NLparameter(cge, es_d[r in regions, g in goods]    == SUB_ELAST[:d]); # Domestic demand aggregation nest (intranational) - substitution elasticity
+@NLparameter(cge, es_f[r in regions, g in goods]    == SUB_ELAST[:f]); # Domestic and foreign demand aggregation nest (international) - substitution elasticity
 
 
 ################
 # VARIABLES
 ################
 
 # Set lower bound
-sv = 0.001
+sv = 0.001                                                                          # (note)
 
 #sectors
-@variable(cge, Y[(r, s) in sub_set_y] >= sv, start = 1);
+@variable(cge, Y[(r, s) in sub_set_y] >= sv, start = 1);                            # (note)
```

Comment:
This is ```1``` most of the time, except for a couple of situations. Note the variable declaration for ```RA```.  

I'm not sure on your second point, will need to look it up. My guess is that it would force to the lower bound, not zero. This is unlikely to create issues for our purposes. However, this could create issues in a dynamic model with a really long time horizon, as the base year benchmark prices of ```1``` will be discounted each year (e.g. ```(1/(1+discount_rate))^time_step```). After 100, 200, 500, 1000, periods this can get really small, but the lower bound could also be scaled proportionally or set to a really low value to begin with, if needed.